tests/unit/pages/Editor.spec.js

```diff
@@ -1485,51 +1430,109 @@ describe("Editor.vue", () => {
-    expect(wrapper.vm.dialogCancelButtonConfig.class).toBe(
+    expect(store.state.dialog.cancelButtonConfig.enabled).toBeTruthy();
+    expect(store.state.dialog.cancelButtonConfig.text).toBe("No");
+    expect(store.state.dialog.cancelButtonConfig.class).toBe(
       "bg-white hover:bg-gray-100 focus:outline-none text-primary"
     );
-    expect(wrapper.vm.optionIndexToDelete).toBe(0);
     expect(wrapper.vm.dialogAction).toBe("deleteOption");
     expect(wrapper.vm.isDialogBoxShown).toBeTruthy();
-    expect(wrapper.find('[data-test="dialogBox"]').exists()).toBeTruthy();
 
-    await wrapper
-      .find('[data-test="dialogBox"]')
-      .find('[data-test="confirmButton"]')
-      .trigger("click");
-    expect(dialogConfirmed).toHaveBeenCalled();
-    expect(confirmDeleteOption).toHaveBeenCalled();
+    expect(wrapper.vm.optionIndexToDelete).toBe(0);
+
+    // simulate clicking the confirm button of the dialog box
+    await simulateConfirmClick(); // deepscan-disable-line
+
+    expect(deleteSelectedOption).toHaveBeenCalled();
     expect(showCannotDeleteOptionDialog).toHaveBeenCalled();
 
-    expect(wrapper.vm.dialogTitle).toBe("Cannot delete the option");
-    expect(wrapper.vm.dialogDescription).toBe(
+    expect(store.state.dialog.title).toBe("Cannot delete the option");
+    expect(store.state.dialog.description).toBe(
       "A question must have at least 2 options"
     );
-    expect(wrapper.vm.dialogConfirmButtonConfig.enabled).toBeTruthy();
-    expect(wrapper.vm.dialogConfirmButtonConfig.text).toBe("Got it");
-    expect(wrapper.vm.dialogConfirmButtonConfig.class).toBe(
+    expect(store.state.dialog.confirmButtonConfig.enabled).toBeTruthy();
+    expect(store.state.dialog.confirmButtonConfig.text).toBe("Got it");
+    expect(store.state.dialog.confirmButtonConfig.class).toBe(
       "bg-primary hover:bg-primary-hover focus:outline-none focus:ring-0"
     );
-    expect(wrapper.vm.dialogCancelButtonConfig.enabled).toBeFalsy();
-    expect(wrapper.vm.dialogCancelButtonConfig.text).toBe("");
-    expect(wrapper.vm.dialogCancelButtonConfig.class).toBe("");
+    expect(store.state.dialog.cancelButtonConfig.enabled).toBeFalsy();
+    expect(store.state.dialog.cancelButtonConfig.text).toBe("");
+    expect(store.state.dialog.cancelButtonConfig.class).toBe("");
+
     expect(wrapper.vm.isDialogBoxShown).toBeTruthy();
-    await wrapper
-      .find('[data-test="dialogBox"]')
-      .find('[data-test="confirmButton"]')
-      .trigger("click");
+    await store.dispatch("dialog/unsetDialogDescription");
+
+    // simulate clicking the confirm button of the dialog box
+    await simulateConfirmClick(); // deepscan-disable-line
+
+    // dialogAction and dialog confirm click status should be reset
+    expect(wrapper.vm.dialogAction).toBeFalsy();
+    expect(wrapper.vm.isDialogConfirmClicked).toBeFalsy();
+  });
+
+  it("cancelling delete option dialog resets index to delete", async () => {
```

Comment:
```suggestion
  it("cancelling delete option dialog resets option's index to delete", async () => {
```
Was wondering for a while what index